src/native/managed/cdacreader/src/Legacy/SOSDacImpl.cs

```diff
@@ -81,9 +82,63 @@ public int GetBreakingChangeVersion()
     public unsafe int GetMethodDescPtrFromFrame(ulong frameAddr, ulong* ppMD) => HResults.E_NOTIMPL;
     public unsafe int GetMethodDescPtrFromIP(ulong ip, ulong* ppMD) => HResults.E_NOTIMPL;
     public unsafe int GetMethodDescTransparencyData(ulong methodDesc, void* data) => HResults.E_NOTIMPL;
-    public unsafe int GetMethodTableData(ulong mt, void* data) => HResults.E_NOTIMPL;
+    public unsafe int GetMethodTableData(ulong mt, DacpMethodTableData* data)
+    {
+        if (mt == 0 || data == null)
+            return HResults.E_INVALIDARG;
+        try
+        {
+            Contracts.IMetadata contract = _target.Contracts.Metadata;
+            Contracts.MethodTableHandle methodTable = contract.GetMethodTableHandle(mt);
+
+            DacpMethodTableData result = default;
+            result.baseSize = contract.GetBaseSize(methodTable);
+            if (contract.IsString(methodTable))
+                result.baseSize -= 2 /*sizeof(WCHAR) */;
```

Comment:
No idea.

@mikem8361 why does the SOS DAC do this:

https://github.com/dotnet/runtime/blob/24b59d85541c72fd55b8d5d0bc9adb84619d6c4f/src/coreclr/debug/daccess/request.cpp#L1799-L1800

Reply:
Maybe better:
```suggestion
                result.baseSize -= sizeof(char);
```

Reply:
The DAC APIs are stuck in time. As you know there are many places where the DAC implementation returns dummy values or constants to keep the DAC API "working". This is another one of those cases. The BaseSize implementation in the runtime changed. but something that consumed DAC API was expecting the old BaseSize value for string and this was added to keep things "working".

For reference, this is the checkin description for the change that introduced this adjustment:

```
	2008/09/25

	Title: New implementation of StringBuilder and improvements in String class
	
	Issue description:
	  1. The existing implementation of StringBuilder uses a String object as
	  its internal buffer. Every time we are short of space, we double the
	  String size and very soon we find ourselves into the large object heap.
	  Since Gen2 collections do not happen as often as Gen0 and Gen1
	  collections, these objects (even if no longer used) remain allocated
	  2. Secondly, this dependency on String necessitates the need to
	  differentiate between arrayLength and stringLength. With the change in 1
	  above, arrayLength is always 1 more than stringLength (the
	  null-terminator) and we can get rid of arrayLength and derive that from
	  stringLength based on need.
	
	Change description:
	  1. This new implementation implements StringBuilder as a linked list of
	  StringBuilder (called chunks). The buffer used internally by the
	  StringBuilder is a char array which is doubled until the chunk size
	  reaches a cap of 8K after which we allocate a new chunk. Almost all the
	  implementation in StringBuilder.cs has been done by Vance.
	  a. StringBuilder.Append faster by 25% on an average
	  b. StringBuilder.ToString is as performant as before
	  2. Remove the bufferLength from the String class. 4 bytes saved per
	  String (except empty Strings where we do padding). So working set gains
	  Different CR'ers have reviewed different sections of the code. Jan and
	  Brian sort of looked at the overall change
```

I hope we will avoid shenanigans like this one going forward by having explicit versioning scheme.

Reply:
Thanks @jkotas. Jeremy and I have been playing a lot of games recently with these sorts of semantics that we respect but are historical artifacts. We often label them with a comment such as below to indicate this is present due to compat reasons.

> `// [Compat] <MINOR DETAIL ON WHAT WE FOUND>`

Reply:
This adjustment is complementary to the inverse adjustment done in the SOS here: https://github.com/dotnet/diagnostics/blob/464e06c9a8602dff44ff5c59e9a0c3777f68e41c/src/SOS/Strike/eeheap.cpp#L447-L450 . In the ideal world, we would delete both.